target/scripts/helpers/postfix.sh

```diff
@@ -91,3 +94,58 @@ function _vhost_ldap_support() {
 #
 # /etc/aliases is handled by `alias.sh` and uses `postalias` to update the Postfix alias database. No need for `postmap`.
 # http://www.postfix.org/postalias.1.html
+
+# Add an option with an argument to Postfix's main configuration file
+# or update an existing option.
+#
+# @param ${1} = option name
+# @param ${2} = value to append when option does not exist yet
+# @param ${3} = replacement string for `sed` to update the configuration
+#               file in case the option (${1}) does exist
+#
+# ## Attention
+#
+# This function is not meant to be called directly from other scripts.
+# There are wrappers like `_append_to_option_in_postfix_main` that use
+# this function under the hood.
+function __add_to_or_update_postfix_main() {
+  if grep -E "^${1:?Option name is required}" "${POSTFIX_MAIN_CF}"; then
+    sed -i -E "${3:?Replacement string is required}" "${POSTFIX_MAIN_CF}"
+  else
+    # We have to use `sed` here, not `echo`, because
+    # the strings are escaped for `sed` already.
+    sed -i -E "\$a${1} = ${2:?String to append is required}" >>"${POSTFIX_MAIN_CF}"
```

Comment:
The `>>` are a typo?

Reply:
> I get that this line is adding the setting + value when the setting does not already exist in the file.
> 
> However, I'm not familiar with the `\$a` portion, what's `sed` doing here with `$a`?

This is telling `sed` to append to the end of the file: `$` match the end, `a` append.

> The `>>` are a typo?

Absolutely! Nice catch.

Reply:
```suggestion
    # We have to use `sed` here, not `echo`, because
    # the strings are escaped for `sed` already.
    sed -i -E "\$a${1} = ${2:?String to append is required}" "${POSTFIX_MAIN_CF}"
```